src/test/test-blockdev-util.c

```diff
@@ -26,6 +33,36 @@ static void test_path_is_encrypted_one(const char *p, int expect) {
         assert_se(expect < 0 || ((r > 0) == (expect > 0)));
 }
 
+TEST(get_block_device) {
+    _cleanup_free_ char *path = NULL, *devnode = NULL;
+    _cleanup_(sd_device_unrefp) sd_device *sd_dev = NULL;
+
+    int r;
+    dev_t devnum;
+
+    ASSERT_OK(mkdtemp_malloc(NULL, &path));
```

Comment:
The created temporary directory should be removed when not necessary.
Please use `_cleanup_(rm_rf_physical_and_freep)` attribute for `path`.
See similar cases in other tests.